src/cmd_parser.cpp

```diff
@@ -573,15 +573,20 @@ bool CmdParser::parseNextCommand()
         children.push_back(e);
       }
       // compute the type of applying the rule
+      std::stringstream ss;
```

Comment:
You should not allocate this by default. You should redo the type checking if it fails.

Reply:
We should just do something like `ExprParser::typeCheckApp` here. I can do this separately if you want.